components/script/fetch.rs

```diff
@@ -232,43 +227,19 @@ pub(crate) fn Fetch(
     // Is `comp` as argument
 
     // Step 9. Let locallyAborted be false.
-    // TODO
     // Step 10. Let controller be null.
-    // TODO
-    // Step 11. Add the following abort steps to requestObject’s signal:
-    // TODO
-    // Step 11.1. Set locallyAborted to true.
-    // TODO
-    // Step 11.2. Assert: controller is non-null.
-    // TODO
-    // Step 11.3. Abort controller with requestObject’s signal’s abort reason.
-    // TODO
-    // Step 11.4. Abort the fetch() call with p, request, responseObject, and requestObject’s signal’s abort reason.
-    // TODO
-
-    // Step 12. Set controller to the result of calling fetch given request and
-    // processResponse given response being these steps:
     let fetch_context = Arc::new(Mutex::new(FetchContext {
         fetch_promise: Some(TrustedPromise::new(promise.clone())),
         response_object: Trusted::new(&*response),
         resource_timing: ResourceFetchTiming::new(timing_type),
+        locally_aborted: false,
     }));
 
-    // Step 12.1. If locallyAborted is true, then abort these steps.
-    // TODO
-    // Step 12.2. If response’s aborted flag is set, then:
-    // TODO
-    // Step 12.2.1. Let deserializedError be the result of deserialize a serialized
-    // abort reason given controller’s serialized abort reason and relevantRealm.
-    // TODO
-    // Step 12.2.2. Abort the fetch() call with p, request, responseObject, and deserializedError.
-    // TODO
-    // Step 12.2.3. Abort these steps.
-    // TODO
+    // Step 11. Add the following abort steps to requestObject’s signal:
+    signal.add(&AbortAlgorithm::Fetch(fetch_context.clone()));
```

Comment:
This doesn't work, because the types don't match:

```
error[E0308]: mismatched types
   --> components/script/fetch.rs:239:39
    |
239 |     signal.add(&AbortAlgorithm::Fetch(fetch_context.clone()));
    |                 --------------------- ^^^^^^^^^^^^^^^^^^^^^ expected `FetchContext`, found `Arc<Mutex<FetchContext>>`
    |                 |
    |                 arguments to this enum variant are incorrect
    |
    = note: expected struct `fetch::FetchContext`
               found struct `std::sync::Arc<std::sync::Mutex<fetch::FetchContext>>`
```

Why does it work for `PipeTo` though, since that's also a struct?